assets/js/modules/optimize/components/common/OptimizeSnippetNotice.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * OptimizeSnippetNotice component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useMemo } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { MODULES_ANALYTICS } from '../../../analytics/datastore/constants';
+import { MODULES_ANALYTICS_4 } from '../../../analytics-4/datastore/constants';
+import { CORE_MODULES } from '../../../../googlesitekit/modules/datastore/constants';
+
+const { useSelect } = Data;
+
+export default function OptimizeSnippetNotice() {
+	const analytics4ModuleConnected = useSelect( ( select ) =>
+		select( CORE_MODULES ).isModuleConnected( 'analytics-4' )
+	);
+
+	const useGA4Snippet = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getUseSnippet()
+	);
+
+	const useUASnippet = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS ).getUseSnippet()
+	);
+
+	const notice = useMemo( () => {
```

Comment:
This `useMemo` is not needed. Let's remove it.

Reply:
Removed the `useMemo` and instead used a more imperative approach. Question, why's the use of `useMemo` bad here? Thanks!

Reply:
@kuasha420 the question here is not why it is bad, the question is why the use of `useMemo` here is needed? The `useMemo` hook is needed to avoid expensive calculations on every render. Translating a string is not an expensive calculation from my point of view, thus we don't need this hook.